wagtail/test/utils/wagtail_tests.py

```diff
@@ -1,12 +1,77 @@
 import warnings
 from contextlib import contextmanager
 
+from bs4 import BeautifulSoup, ResultSet, Tag
 from django import VERSION as DJANGO_VERSION
 from django.contrib.auth import get_user_model
+from django.http import HttpResponse
+from django.test.client import Client as DjangoTestClient
 from django.test.testcases import assert_and_parse_html
+from django.utils.functional import SimpleLazyObject
+
+
+class SoupHttpResponse(HttpResponse):
+    """Subclass for type-hinting purposes.
+
+    Use it like this for type-hinting:
+        response: SoupHttpResponse = self.client.get("/your-url/")
+
+        # Wonderful type hinting!
+        response.soup.select(...)
+    """
+
+    soup: BeautifulSoup
+
+    def select(self, selector: str) -> ResultSet[Tag]:
+        pass
+
+    def select_one(self, selector: str) -> (Tag | None):
+        pass
+
+
+class BS4SoupClient(DjangoTestClient):
+    """Convenience wrapper around Django's Client and BeautifulSoup's SoupSieve
+
+    The intend of this wrapper is to make it easier to check the html of a response.
+
+    Usage example:
+
+        client = BS4SoupClient()
+        response = client.get("/good-vibes/")
+
+        # Select a HTML <span class="good-vibes> in the response
+        good_vibes = response.select("span.good-vibes")
+
+        # Assert the is only one element on the page that matches the selector
+        self.assertEqual(len(soup), 1)
+    """
+
+    def _get_soup(self, response):
+        content_type = response.headers.get("Content-Type", None)
+        if "text/html" in content_type:
+            return BeautifulSoup(response.content, "html5lib")
+        raise ValueError(
+            "Cannot use BeautifulSoup, the response content type is not HTML"
+        )
+
+    def request(self, **request) -> SoupHttpResponse:
+        response = super().request(**request)
+        # Make the soup available lazily. We will only incur the cost of instantiating
+        # the soup upon the initial access of this attribute.
+        response.soup = SimpleLazyObject(lambda: self._get_soup(response))
+
+        # Alias these BeautifulSoup functions the the response for convenience
+        response.select = lambda selector: response.soup.select(selector)
+        response.select_one = lambda selector: response.soup.select_one(selector)
```

Comment:
The benefit of `response.select` vs `response.soup.select` doesn't feel very high to me to warrant extra properties.